### PR TITLE
fix: nullptr access for logger in master

### PR DESCRIPTION
**Pull request by EmosewaMC**
Tested that my characters position and data saves on shutdown now and that master no longer hangs on shutdown

### CODE REVIEW

**Review by aronwk-aaron — APPROVED** [review]

File: dWorldServer/WorldServer.cpp [hunk 1]
```diff
@@ -1291,14 +1291,17 @@ void WorldShutdownProcess(uint32_t zoneId) {
...
 	Game::lastSignal = -1;
 #ifndef DARKFLAME_PLATFORM_WIN32
-	if (Game::ShouldShutdown() || worldShutdownSequenceComplete)
+	if (shouldShutdown)
 #endif
 	{
 		return;
 	}
 
+	if (!Game::logger) return;
```
Comment (Xiphoseer) [inline dWorldServer/WorldServer.cpp:1303]: Can we printf when this happens? This still method still feels weird in general.

**Review by Xiphoseer — APPROVED** [review]

**Comment by Xiphoseer** [comment]
Was this on windows? Preferably we'd sort out differences instead of silently returning on weird conditions all over the place.

**Comment by aronwk-aaron** [comment]
> Was this on windows? Preferably we'd sort out differences instead of silently returning on weird conditions all over the place.

If it was with windows, it's to do with the signal spawning a separate thread to handle the shutdown, which id a windows specific thing

**Comment by EmosewaMC** [comment]
> Was this on windows? Preferably we'd sort out differences instead of silently returning on weird conditions all over the place.

This is a non windows issue and causes master to hang indefinitely on my wsl instance.

**Comment by EmosewaMC** [comment]
Windows must have that define.  No other way around it

**Pull request merged**